ydb/core/blobstorage/pdisk/blobstorage_pdisk_completion_impl.cpp

```diff
@@ -11,6 +11,55 @@ namespace NPDisk {
 // Completion actions
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Chunk write completion actions
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+
+TCompletionChunkWritePart::TCompletionChunkWritePart(TChunkWritePiece* piece, TCompletionChunkWrite* cumulativeCompletion)
+    : TCompletionAction()
+    , PDisk(piece->PDisk)
+    , PieceShift(piece->PieceShift)
+    , PieceSize(piece->PieceSize)
+    , CumulativeCompletion(cumulativeCompletion)
+    , Span(piece->Span.CreateChild(TWilson::PDiskDetailed, "PDisk.ChunkWritePiece.CompletionPart"))
+{}
+
+TCompletionChunkWritePart::~TCompletionChunkWritePart() {
+    if (CumulativeCompletion) {
+        CumulativeCompletion->RemovePart(PDisk->PCtx->ActorSystem);
+    }
+}
+
+void TCompletionChunkWritePart::Exec(TActorSystem *actorSystem) {
+    Span.Event("PDisk.CompletionChunkWritePart.Exec");
+
+    Y_VERIFY(actorSystem);
+    Y_VERIFY(CumulativeCompletion);
+    if (TCompletionAction::Result != EIoResult::Ok) {
+        Release(actorSystem);
+        return;
+    }
+    Y_UNUSED(PieceShift);
+    Y_UNUSED(PieceSize);
+    
+
+    // double deviceTimeMs = HPMilliSecondsFloat(GetTime - SubmitTime);
```

Comment:
uncomment once i figure out what to do with racing access to ChunkWrite->Orbit. Should i create new orbits or use mutex or do not track this at all